homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,317 @@
+        self._temperature_unit = hass.config.units.temperature_unit
+
+        # Setup supported hvac modes
+        self._hvac_modes = [HVAC_MODE_OFF]
+
+        # Add supported lyric thermostat features
+        if "Heat" in device.allowedModes:
+            self._hvac_modes.append(HVAC_MODE_HEAT)
+
+        if "Cool" in device.allowedModes:
+            self._hvac_modes.append(HVAC_MODE_COOL)
+
+        if "Heat" in device.allowedModes and "Cool" in device.allowedModes:
+            self._hvac_modes.append(HVAC_MODE_HEAT_COOL)
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_thermostat",
+            device.name,
+            "mdi:thermostat",
+        )
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        return self._temperature_unit
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    return device.indoorTemperature
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the hvac mode."""
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    return HVAC_MODES[device.changeableValues.mode]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """List of available hvac modes."""
+        return self._hvac_modes
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    if not device.hasDualSetpointStatus:
+                        return device.changeableValues.heatSetpoint
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
+                    if device.hasDualSetpointStatus:
+                        return device.changeableValues.coolSetpoint
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        for location in self.coordinator.data.locations:
+            for device in location.devices:
+                if device.macID == self._device.macID:
```

Comment:
The device id search pattern is repeated many times here.  Can we make this a bit more dry?

Please have the coordinator return the data from the request instead of the object itself.

It probably makes sense to reformat the data when it comes back from the library to create a dict of devices so all of these properties and functions can avoid the multi level linear searches.